doc/over_sampling.rst

```diff
@@ -0,0 +1,227 @@
+
+Ill-posed examples
+------------------
+
+While the :class:`RandomOverSampler` is over-sampling by repeating some of the
+original samples, :class:`SMOTE` and :class:`ADASYN` generate new samples in by
+interpolation. However, the samples used to interpolate/generate new synthetic
+samples differ. In fact, :class:`ADASYN` focuses on generating samples next to
+the original samples which are wrongly classified using a k-Nearest Neighbors
+classifier while the basic implementation of :class:`SMOTE` will not make any
+distinction between easy and hard samples to be classified using the nearest
+neighbors rule. Therefore, the decision function found during training will be
+different between the different algorithms.
+
+.. image:: ./auto_examples/over-sampling/images/sphx_glr_plot_comparison_over_sampling_004.png
+   :target: ./auto_examples/over-sampling/plot_comparison_over_sampling.html
+   :align: center
+
+The sampling particularities of these two algorithms can lead to some peculiar
+behavior as shown below.
+
+.. image:: ./auto_examples/over-sampling/images/sphx_glr_plot_comparison_over_sampling_005.png
+   :target: ./auto_examples/over-sampling/plot_comparison_over_sampling.html
+   :scale: 60
+   :align: center
+
+SMOTE variants
+--------------
+
+SMOTE might connect inliers and outliers while ADASYN might focus solely on
+outliers which, in both cases, might lead to a sub-optimal decision
+function. In this regard, SMOTE offers three additional options to generate
+samples. Those methods focus on samples near of the border of the optimal
+decision function and will generate samples in the opposite direction of the
+nearest neighbors class. Those variants are presented in the figure below.
+
+.. image:: ./auto_examples/over-sampling/images/sphx_glr_plot_comparison_over_sampling_006.png
+   :target: ./auto_examples/over-sampling/plot_comparison_over_sampling.html
+   :scale: 60
+   :align: center
+
+
+The parameter ``kind`` is controlling this feature and the following types are
+available: (i) ``'borderline1'``, (ii) ``'borderline2'``, and (iii) ``'svm'``::
+
+  >>> from imblearn.over_sampling import SMOTE, ADASYN
+  >>> X_resampled, y_resampled = SMOTE(kind='borderline1').fit_sample(X, y)
+  >>> print(Counter(y_resampled))
+  Counter({2: 4674, 1: 4674, 0: 4674})
+
+See :ref:`sphx_glr_auto_examples_over-sampling_plot_comparison_over_sampling.py`
+to see a comparison between the different over-sampling methods.
+
+Mathematical formulation
+========================
+
+Sample generation
+-----------------
+
+Both SMOTE and ADASYN use the same algorithm to generate new
+samples. Considering a sample :math:`x_i`, a new sample :math:`x_{new}` will be
+generated considering its k neareast-neighbors (corresponding to
+``k_neighbors``). For instance, the 3 nearest-neighbors are included in the
+blue circle as illustrated in the figure below. Then, one of these
+nearest-neighbors :math:`x_{zi}` will be selected and a sample will be
```

Comment:
using will on the previous sentence is fine. 'cos is something that would appear as a consequence. But here confuses me and I had to read twice (acutally more :P)
`is selected and the new sample is generated as follows:`